### ISSUE TITLE
valid( Joi.override, "string" )

### ISSUE THREAD

**Issue by Kaseano** [mrjono1/joi-to-typescript#229]
Ran into an issue using Joi.override with AnySchema..valid()

**demoSchema.ts**
```typescript
import Joi from "joi";

export default Joi.object({

	// OK:
	// foo: Joi.string().valid("bar").valid("baz")

	// OK:
	// foo: Joi.string().valid("bar").valid("baz", "qux")

	// NOT OK:
	foo: Joi.string().valid("bar").valid(Joi.override, "baz", "qux")

	// TypeError: value.replace is not a function
	// at toStringLiteral (\node_modules\joi-to-typescript\dist\main\utils.js:26:22)

})
.meta({ className: "demo" })
```

Test code used to confirm that the joi is valid:
**testDemo.ts**
```typescript
import demo from "./demoSchema"

// confirm that joi is valid:
// yarn ts-node ./test/sandbox/testDemo.ts
{
	// should err
	const { value, warning, error } = demo.validate({ foo: "bar" })
	if (!error) throw new Error("should have errored")
} {
	// should not err
	const { value, warning, error } = demo.validate({ foo: "baz" })
	if (error) throw new Error("should not have errored")
}
console.log("done")
```

**Comment by mrjono1** [comment]
I haven't tried Joi.override before

**Comment by Kaseano** [comment]
It's useful sometimes.  I made a pull request that'll hopefully explain it better.

**Comment by mrjono1** [comment]
https://github.com/mrjono1/joi-to-typescript/pull/230

**Comment by mrjono1** [comment]
I have merged and released your changes they will be available on version 4.0.2
